cloud/ec2.go

```diff
@@ -297,71 +278,112 @@ func (m *ec2Manager) spawnOnDemandHost(ctx context.Context, h *host.Host, ec2Set
-		spotRequest.LaunchSpecification.NetworkInterfaces = []*ec2.InstanceNetworkInterfaceSpecification{
-			&ec2.InstanceNetworkInterfaceSpecification{
+		launchTemplate.NetworkInterfaces = []*ec2.LaunchTemplateInstanceNetworkInterfaceSpecificationRequest{
+			{
 				AssociatePublicIpAddress: aws.Bool(true),
 				DeviceIndex:              aws.Int64(0),
 				Groups:                   ec2Settings.getSecurityGroups(),
 				SubnetId:                 &ec2Settings.SubnetId,
 			},
 		}
 		if ec2Settings.IPv6 {
-			spotRequest.LaunchSpecification.NetworkInterfaces[0].SetIpv6AddressCount(1).SetAssociatePublicIpAddress(false)
+			launchTemplate.NetworkInterfaces[0].SetIpv6AddressCount(1).SetAssociatePublicIpAddress(false)
 		}
 	} else {
-		spotRequest.LaunchSpecification.SecurityGroups = ec2Settings.getSecurityGroups()
+		launchTemplate.SecurityGroups = ec2Settings.getSecurityGroups()
+	}
+
+	userData, err := bootstrapUserData(ctx, evergreen.GetEnvironment(), h, ec2Settings.UserData)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not add bootstrap script to user data")
 	}
+	ec2Settings.UserData = userData
 
 	if ec2Settings.UserData != "" {
-		expanded, err := m.expandUserData(ec2Settings.UserData)
+		var expanded string
+		expanded, err = m.expandUserData(ec2Settings.UserData)
 		if err != nil {
 			return nil, errors.Wrap(err, "problem expanding user data")
 		}
 		userData := base64.StdEncoding.EncodeToString([]byte(expanded))
-		spotRequest.LaunchSpecification.UserData = &userData
+		launchTemplate.UserData = &userData
 	}
 
-	userData, err := bootstrapUserData(ctx, evergreen.GetEnvironment(), h, ec2Settings.UserData)
+	createTemplateResponse, err := m.client.CreateLaunchTemplate(ctx, &ec2.CreateLaunchTemplateInput{
+		LaunchTemplateData: launchTemplate,
+		// mandatory field may only contain letters, numbers, and the following characters: - ( ) . / _
+		LaunchTemplateName: aws.String(fmt.Sprintf("%s", util.RandomString())),
+	})
 	if err != nil {
-		return nil, errors.Wrap(err, "could not add bootstrap script to user data")
+		return nil, errors.Wrap(err, "can't upload config template to AWS")
+	}
+	err = validateCreateTemplateResponse(createTemplateResponse)
+	if err != nil {
+		return nil, errors.Wrap(err, "invalid create template response")
+	}
+	templateID = createTemplateResponse.LaunchTemplate.LaunchTemplateId
+
+	// Create a fleet with a single spot instance from the launch template
+	createFleetInput := &ec2.CreateFleetInput{
+		LaunchTemplateConfigs: []*ec2.FleetLaunchTemplateConfigRequest{
+			{
+				LaunchTemplateSpecification: &ec2.FleetLaunchTemplateSpecificationRequest{
+					LaunchTemplateId: templateID,
+					Version:          aws.String(strconv.Itoa(int(*createTemplateResponse.LaunchTemplate.LatestVersionNumber))),
+				},
+			},
+		},
+		TargetCapacitySpecification: &ec2.TargetCapacitySpecificationRequest{
+			TotalTargetCapacity:       aws.Int64(1),
+			DefaultTargetCapacityType: aws.String("spot"),
```

Comment:
Is "spot" a constant in the `aws` package?

Reply:
yes. 
done.